scripts/build/gen_handles.py

```diff
@@ -372,39 +372,52 @@ def main():
             inj_paths = []
             sup_paths = []
             hdls = []
+            hdls_temp = []
 
             sn = hs.node
+            hdls_temp.clear()
             if sn:
-                hdls.extend(dn.__device.dev_handle for dn in sn.__depends)
+                hdls_temp.extend(dn.__device.dev_handle for dn in sn.__depends)
+                hdls_temp.sort()
+                hdls.extend(hdls_temp)
                 for dn in sn.depends_on:
                     if dn in sn.__depends:
                         dep_paths.append(dn.path)
                     else:
                         dep_paths.append('(%s)' % dn.path)
             # Force separator to signal start of injected dependencies
             hdls.append(DEVICE_HANDLE_SEP)
+            hdls_temp.clear()
             for inj in hs.dev_injected:
                 if inj not in edt.dep_ord2node:
                     continue
                 expected = edt.dep_ord2node[inj]
                 if expected in used_nodes:
                     inj_paths.append(expected.path)
-                    hdls.append(expected.__device.dev_handle)
+                    hdls_temp.append(expected.__device.dev_handle)
+                hdls_temp.sort()
+                hdls.extend(hdls_temp)
 
             # Force separator to signal start of supported devices
             hdls.append(DEVICE_HANDLE_SEP)
+            hdls_temp.clear()
             if len(hs.dev_sups) > 0:
                 for dn in sn.required_by:
                     if dn in sn.__supports:
                         sup_paths.append(dn.path)
                     else:
                         sup_paths.append('(%s)' % dn.path)
-                hdls.extend(dn.__device.dev_handle for dn in sn.__supports)
+                hdls_temp.extend(dn.__device.dev_handle for dn in sn.__supports)
+                hdls_temp.sort()
+                hdls.extend(hdls_temp)
 
+            hdls_temp.clear()
             if args.num_dynamic_devices != 0:
                 pm = pm_devices.get(dev.obj_pm)
                 if pm and pm.is_domain():
-                    hdls.extend(DEVICE_HANDLE_NULL for dn in range(args.num_dynamic_devices))
+                    hdls_temp.extend(DEVICE_HANDLE_NULL for dn in range(args.num_dynamic_devices))
+                    hdls_temp.sort()
+                    hdls.extend(hdls_temp)
 
```

Comment:
Could we just do a sort once at the end?

Reply:
The device handles include multiple sections.  See the description [here](https://github.com/zephyrproject-rtos/zephyr/blob/e1836718082c8b24a31372373b69d6b52546b09b/include/zephyr/device.h#L884)